src/test/java/com/yourcompany/TestRules/RetryRule.java

```diff
@@ -31,17 +25,16 @@ public void evaluate() throws Throwable {
                 Throwable caughtThrowable = null;
 
                 // implement retry logic here
-                while (retryCount.getAndDecrement() > 0) {
+                int i = retryCount;
+                while (i-- > 0) {
                     try {
                         base.evaluate();
                         return;
                     } catch (Throwable t) {
-                        if (retryCount.get() > 0 && description.getAnnotation(Retry.class)!= null) {
```

Comment:
Wouldn't adding a  "=" to the first comparison fix or pulling out the assignment to caughtThrowable from within the if statement?